### PR TITLE
MWSS Pay Pattern summary

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
I have added the Summary of selected Pay Patterns.
Also this PR includes the ability to route between groups

### How to review 
Check that the behaviour of the MWSS Pay Pattern summary matches the slides

### CODE REVIEW

**Review by adoublebarrel — APPROVED** [review]
I'm still not a fan of using Jinja filters like this but it works.

**Review by yoganandkunche — APPROVED** [review]

File: app/submitter/converter.py [hunk 1]
```diff
@@ -94,6 +94,9 @@ def convert_answers_to_data(answer_store, questionnaire_json, routing_path):
 
             if answer_schema is not None and value is not None and 'parent_answer_id' not in answer_schema:
                 if answer_schema['type'] != 'Checkbox' or any('q_code' not in option for option in answer_schema['options']):
+                    if 'q_code' not in answer_schema:
```
Comment (dwyeradam) [inline app/submitter/converter.py]: Are we happy that we could lose data if q_codes in schemas have not been defined?

File: app/submitter/converter.py [hunk 2]
```diff
@@ -94,6 +94,9 @@ def convert_answers_to_data(answer_store, questionnaire_json, routing_path):
...
+                        continue
+
```
Comment (dwyeradam) [inline app/submitter/converter.py:99]: Are we happy that we could lose data if q_codes in schemas have not been defined?

Comment (jonnyshaw89) [reply app/submitter/converter.py:99]: Yes, given that this isn't enforced in the schema then we shouldn't enforce it in the code.

**Review by dwyeradam — APPROVED** [review]
Based on the fact we no longer enforce full q_code use across the schema.

**Review by ajmaddaford — CHANGES_REQUESTED** [review]
Leaving a blocking review as we don't want this merged yet.

File: app/jinja_filters.py [hunk 3]
```diff
@@ -102,14 +102,26 @@ def format_household_member_name(names):
 
 
 @blueprint.app_template_filter()
+def format_summary(list_items):
```
Comment (ajmaddaford) [inline app/jinja_filters.py]: I think this should be `format_list` rather than `format_summary`

File: data/schema/schema_v1.json [hunk 4]
```diff
@@ -162,10 +162,25 @@
...
+                "type": "string"
+              },
               "when": {
                 "$ref": "#/definitions/when"
               }
             },
+            "oneOf": [
+              {
+                "required": [
+                  "id"
```
Comment (ajmaddaford) [inline data/schema/schema_v1.json:175]: I think we should consider renaming `id` to `block` now that we have `group`

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: data/schema/schema_v1.json (hunk 4 above)
Comment (jonnyshaw89) [reply data/schema/schema_v1.json:175]: Added Card https://trello.com/c/zAJUVc8q/335-rename-id-in-routingrules-to-block

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**